File: app/src/shemerly/java/com/quran/labs/androidquran/util/ShemerlyPageProvider.java
package com.quran.labs.androidquran.util;

public class ShemerlyPageProvider implements QuranScreenInfo.PageProvider {

  @Override
  public String getWidthParameter() {
    return "1200";
  }

  @Override
  public String getTabletWidthParameter() {
    // use the same size for tablet landscape
    return getWidthParameter();
  }

  @Override
  public void setOverrideParameter(String parameter) {
    // override parameter is irrelevant for shemerly pages
  }
}

File: common/pageinfo/src/main/java/com/quran/data/page/provider/impl/ShemerlyPageProvider.kt
package com.quran.data.page.provider.impl

import com.quran.data.page.provider.PageProvider

internal class ShemerlyPageProvider : PageProvider {
  private val baseUrl = "http://android.quran.com/data/shemerly"

  override fun getImageVersion() = 1

  override fun getImageUrl() = "$baseUrl/"

  override fun getImageZipUrl() = "$baseUrl/zips/"

  override fun getPatchBaseUrl() = "$baseUrl/patches/v"

  override fun getAyahInfoUrl() = "$baseUrl/databases/ayahinfo/"

  override fun getAudioDirectoryName() = "audio"

  override fun getDatabaseDirectoryName() = "databases"

  override fun getAyahInfoDirectoryName() = "shemerly/" + getDatabaseDirectoryName()

  override fun getImagesDirectoryName() = "shemerly"

  override fun getWidthParameter() = "1200"

  override fun getTabletWidthParameter(): String {
    // use the same size for tablet landscape
    return getWidthParameter()
  }

  override fun setOverrideParameter(parameter: String) {
    // override parameter is irrelevant for shemerly pages
  }
}
